app/controllers/shops/products_controller.rb

```diff
@@ -24,9 +25,29 @@ def create
     end
   end
 
+  def edit; end
+
+  def update
+    if @product.update_attributes product_params
+      flash[:success] = t "shop.product.update.update_success"
+      redirect_to shops_products_path
+    else
+      flash[:danger] = t "shop.product.update.update_fail"
+      render :edit
+    end
+  end
+
   private
 
-  def create_product_params
-    params.require(:product).permit Product::CREATE_PRODUCT_PARAMS
+  def load_product
+    @product = Product.find_by slug: params[:slug]
+    return unless @product.blank?
```

Comment:
```suggestion
    return if @product
```
xong rails tut rồi mà vẫn... (tat)

Reply:
dạ vâng e quên mất @@@